x/crosschain/keeper/evm_hooks_test.go

```diff
@@ -2,95 +2,769 @@ package keeper_test
 		require.ErrorContains(t, err, "ParseZRC20WithdrawalEvent: invalid amount")
 	})
-	t.Run("invalid chain ID", func(t *testing.T) {
-		btcMainNetWithdrawalEvent, err := crosschainkeeper.ParseZRC20WithdrawalEvent(*GetValidSampleBlock(t).Logs[3])
+	t.Run("unable to validate a event with an invalid chain ID", func(t *testing.T) {
+		btcMainNetWithdrawalEvent, err := crosschainkeeper.ParseZRC20WithdrawalEvent(*GetValidZRC20WithdrawToBTC(t).Logs[3])
 		require.NoError(t, err)
 		err = crosschainkeeper.ValidateZrc20WithdrawEvent(btcMainNetWithdrawalEvent, common.BtcTestNetChain().ChainId)
 		require.ErrorContains(t, err, "address is not for network testnet3")
 	})
-	t.Run("invalid address type", func(t *testing.T) {
-		btcMainNetWithdrawalEvent, err := crosschainkeeper.ParseZRC20WithdrawalEvent(*GetValidSampleBlock(t).Logs[3])
+	t.Run("unable to validate a event with an invalid address type", func(t *testing.T) {
+		btcMainNetWithdrawalEvent, err := crosschainkeeper.ParseZRC20WithdrawalEvent(*GetValidZRC20WithdrawToBTC(t).Logs[3])
 		require.NoError(t, err)
 		btcMainNetWithdrawalEvent.To = []byte("1EYVvXLusCxtVuEwoYvWRyN5EZTXwPVvo3")
 		err = crosschainkeeper.ValidateZrc20WithdrawEvent(btcMainNetWithdrawalEvent, common.BtcTestNetChain().ChainId)
 		require.ErrorContains(t, err, "decode address failed: unknown address type")
 	})
 }
-func TestParseZRC20WithdrawalEvent(t *testing.T) {
-	t.Run("invalid address", func(t *testing.T) {
-		for i, log := range GetSampleBlockLegacyToAddress(t).Logs {
-			event, err := crosschainkeeper.ParseZRC20WithdrawalEvent(*log)
-			if i < 3 {
+
+func TestKeeper_ProcessZRC20WithdrawalEvent(t *testing.T) {
+	t.Run("successfully process ZRC20Withdrawal to BTC chain", func(t *testing.T) {
+		k, ctx, sdkk, zk := keepertest.CrosschainKeeper(t)
+		k.GetAuthKeeper().GetModuleAccount(ctx, fungibletypes.ModuleName)
+
+		chain := common.BtcMainnetChain()
+		chainID := chain.ChainId
+		setSupportedChain(ctx, zk, chainID)
+		SetupStateForProcessLogs(t, ctx, k, zk, sdkk, chain)
+
+		event, err := crosschainkeeper.ParseZRC20WithdrawalEvent(*GetValidZRC20WithdrawToBTC(t).Logs[3])
+		require.NoError(t, err)
+		zrc20 := setupGasCoin(t, ctx, zk.FungibleKeeper, sdkk.EvmKeeper, chainID, "bitcoin", "BTC")
+		event.Raw.Address = zrc20
+		emittingContract := sample.EthAddress()
+		txOrigin := sample.EthAddress()
+		tss := sample.Tss()
+
+		err = k.ProcessZRC20WithdrawalEvent(ctx, event, emittingContract, txOrigin.Hex(), tss)
+		require.NoError(t, err)
+		cctxList := k.GetAllCrossChainTx(ctx)
+		require.Len(t, cctxList, 1)
+		require.Equal(t, "bc1qysd4sp9q8my59ul9wsf5rvs9p387hf8vfwatzu", cctxList[0].GetCurrentOutTxParam().Receiver)
+		require.Equal(t, emittingContract.Hex(), cctxList[0].InboundTxParams.Sender)
+		require.Equal(t, txOrigin.Hex(), cctxList[0].InboundTxParams.TxOrigin)
+	})
+	t.Run("successfully process ZRC20Withdrawal to ETH chain", func(t *testing.T) {
```

Comment:
```suggestion
	})
	
	t.Run("successfully process ZRC20Withdrawal to ETH chain", func(t *testing.T) {
```
Can we add a space between each test case?

Reply:
837a063a193fcb2066c49b150b332af05d285887